File: solved-cs50-psets/2019-fall/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

int main(void)
{
    // Prompting the user for a text
    string t = get_string("Text: ");
    // Vsriables for all the letters, words and sentences in a text
    int all_letters = 0, all_words = 1, all_sentences = 0, n = strlen(t);
    // Iterating over every character in a text
    for (int i = 0; i < n; i++)
    {
        if ((t[i] >= 'a' && t[i] <= 'z') || (t[i] >= 'A' && t[i] <= 'Z'))
        {
            all_letters++;
        }
        else if ((t[i] == ' ') && (t[i + 1] != ' '))
        {
            all_words++;
        }
        else if ((t[i] == '.') || (t[i] == '?') || (t[i] == '!'))
        {
            all_sentences++;
        }
    }
    // Making sure that there is no whitespaces after a last word with trailing characters
    // and that there is no white space before a first word
    if ((t[n - 1] == ' ') || (t[n - 2] == ' ') || (t[0] == ' '))
    {
        all_words -= 1;
    }

    float avgwords, l, s;
    // Different approaches for calculating L and S values
    if (all_words < 100)
    {
        avgwords = 100 / (float) all_words;
        l = all_letters * avgwords;
        s = all_sentences * avgwords;
    }
    else
    {
        avgwords = (float) all_words / 100;
        l = all_letters / avgwords;
        s = all_sentences / avgwords;
    }
    // Finding a grade level
    float index = 0.0588 * l - 0.296 * s - 15.8;
    // Printing a grade level according to the index
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    if (index >= 1)
    {
        int rounded = round(index);

        if (rounded >= 1 && rounded < 16)
        {
            printf("Grade %i\n", rounded);
        }
        else
        {
            printf("Grade 16+\n");
        }
    }
}

File: solved-cs50-psets/2019-fall/pset6/readability/readability.py
import re
from cs50 import get_string
# The program computes the approximate grade level needed to comprehend some text
user_inp = get_string("Text: ")

all_let, all_words, all_sent = 0, 1, 0
length = len(user_inp)

for i in range(length):
    if user_inp[i].isalpha():
        all_let += 1
    elif user_inp[i] == " " and user_inp[i + 1] != " ":
        all_words += 1
    elif user_inp[i] == "." or user_inp[i] == "!" or user_inp[i] == "?":
        all_sent += 1

if user_inp[length - 1] == " " or user_inp[length - 2] == " ":
    all_words -= 1

if all_words < 100:
    avgwords = 100 / all_words
    l = all_let * avgwords
    s = all_sent * avgwords
else:
    avgwords = all_words / 100
    l = all_let / avgwords
    s = all_sent / avgwords

index = 0.0588 * l - 0.296 * s - 15.8

if index < 1:
    print("Before Grade 1")
else:
    grade = round(index)

    if grade >= 1 and grade < 16:
        print(f"Grade {grade}")
    else:
        print("Grade 16+")